test/format-test.cc

```diff
@@ -771,7 +771,10 @@ TEST(FormatterTest, SpaceSign) {
 
 TEST(FormatterTest, SignNotTruncated) {
   wchar_t format_str[] = {L'{', L':',
-                          '+' | (1 << fmt::detail::num_bits<char>()), L'}', 0};
+                          '+' | static_cast<wchar_t>(
+                            1 << fmt::detail::num_bits<char>() 
+                            & static_cast<typename std::make_unsigned<int>::type >(-1) ),
```

Comment:
The second static cast and & are not needed, please remove.

Reply:
May I ask if it's unneccessary always or just in this case?

Reply:
I don't see why it was added in the first place here. It has no effect.